### PR TITLE
Cleanup compilations & tests

### PR DESCRIPTION
**Pull request by lxn2**
- Changed amalgamation Makefile to be flexible with different flags so we can compile the following:
1. make USE_BLAS=atlas
2. make USE_BLAS=atlas MIN=1
3. make USE_BLAS=atlas MSHADOW_USE_MKL=1
4. make USE_BLAS=atlas MSHADOW_USE_CUDA=1
5. make USE_BLAS=atlas DISABLE_OPENMP=0

- Cleaned up Dockerfile and Jenkins script

### CODE REVIEW

**Comment by lxn2** [comment]
Test results:

    [make USE_BLAS=atlas](http://jenkins-master-elb-1979848568.us-east-1.elb.amazonaws.com/job/mxnet-amalgamation/16/FLAG=1,label=testgpu/console)
    [make USE_BLAS=atlas MIN=1](http://jenkins-master-elb-1979848568.us-east-1.elb.amazonaws.com/job/mxnet-amalgamation/16/FLAG=2,label=testgpu/console)
    [make USE_BLAS=atlas MSHADOW_USE_MKL=1](http://jenkins-master-elb-1979848568.us-east-1.elb.amazonaws.com/job/mxnet-amalgamation/16/FLAG=3,label=testgpu/console)
    [make USE_BLAS=atlas MSHADOW_USE_CUDA=1](http://jenkins-master-elb-1979848568.us-east-1.elb.amazonaws.com/job/mxnet-amalgamation/FLAG=4,label=testgpu/16/console)
    [make USE_BLAS=atlas DISABLE_OPENMP=0](http://jenkins-master-elb-1979848568.us-east-1.elb.amazonaws.com/job/mxnet-amalgamation/FLAG=5,label=testgpu/16/console)

**Comment by piiswrong** [comment]
Can one of the admins verify this patch?

File: amalgamation/Makefile [hunk 1]
```diff
@@ -24,15 +24,25 @@ ifndef EMCC
...
 
 DEFS+=-DMSHADOW_USE_CUDA=0 -DMSHADOW_USE_MKL=0 -DMSHADOW_RABIT_PS=0 -DMSHADOW_DIST_PS=0 -DDMLC_LOG_STACK_TRACE=0
-DEFS+=-DMSHADOW_FORCE_STREAM -DMXNET_USE_OPENCV=0 -DMXNET_PREDICT_ONLY=1 -DDISABLE_OPENMP=1
+DEFS+=-DMSHADOW_FORCE_STREAM -DMXNET_USE_OPENCV=0 -DMXNET_PREDICT_ONLY=1
 CFLAGS=-std=c++11 -Wno-unknown-pragmas -Wall $(DEFS)
 ifneq ($(MIN), 1)
 	CFLAGS += -I${OPENBLAS_ROOT} -I${OPENBLAS_ROOT}/include
-	LDFLAGS+= -L${OPENBLAS_ROOT} -L${OPENBLAS_ROOT}/lib -lopenblas
+	LDFLAGS+= -L${OPENBLAS_ROOT} -L${OPENBLAS_ROOT}/lib
+        ifeq ($(USE_BLAS), openblas)
```
Comment (piiswrong) [inline amalgamation/Makefile]: document USE_BLAS

Comment (lxn2) [inline amalgamation/Makefile]: done

**Pull request merged**